streams/test-utils/src/main/java/org/apache/kafka/streams/internals/MockKeyValueStore.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.internals;
```

Comment:
I did not have time to read the KIP yet, but if we add public API, it should be go into package `internals` that contains classed for non-public stuff.

We should rather create a new package `org.apache.kafka.streams.state` and put the new mocks there (same for the factory). Can you update the KIP accordingly. I'll try to review the KIP if I find time.